test/e2e/framework/framework.go

```diff
@@ -0,0 +1,333 @@
+	if TestContext.DeleteNamespace && (TestContext.DeleteNamespaceOnFailure || !CurrentGinkgoTestDescription().Failed) {
+		for _, ns := range f.namespacesToDelete {
+			By(fmt.Sprintf("Destroying namespace %q for this suite.", ns.Name))
+			if err := f.DeleteNamespace(ns.Name, 5*time.Minute); err != nil {
+				nsDeletionErrors[ns.Name] = err
+			}
+		}
+	}
+
+	for _, backupID := range f.BackupIDs {
+		By(fmt.Sprintf("Deleting backups %q", backupID))
+		ctx := context.Background()
+		f.BlockStorageClient.DeleteVolumeBackup(ctx, coreOCI.DeleteVolumeBackupRequest{VolumeBackupId: &backupID})
+	}
+
+	// if we had errors deleting, report them now.
+	if len(nsDeletionErrors) != 0 {
+		messages := []string{}
+		for namespaceKey, namespaceErr := range nsDeletionErrors {
+			messages = append(messages, fmt.Sprintf("Couldn't delete ns: %q: %s (%#v)", namespaceKey, namespaceErr, namespaceErr))
+		}
+		Failf(strings.Join(messages, ","))
+	}
+
+	f.ProvisionerBlockInstalled = false
+	f.ProvisionerFSSInstalled = false
+}
+
+func (f *Framework) createStorageClient() coreOCI.BlockstorageClient {
+	By("Creating an OCI block storage client")
+	configPath := f.CheckEnvVar(OCIConfigVar)
+
+	file, err := os.Open(configPath)
+	if err != nil {
+		Failf("Unable to load volume provisioner configuration file: %v", configPath)
+	}
+	defer file.Close()
+	cfg, err := client.LoadConfig(file)
+	if err != nil {
+		Failf("Unable to load volume provisioner configuration file %q: %v", file, err)
+	}
+	config, err := f.newConfigurationProvider(cfg)
+	if err != nil {
+		Failf("Unable to load volume provisioner configuration file %q: %v", cfg, err)
+	}
+	blockStorageClient, err := coreOCI.NewBlockstorageClientWithConfigurationProvider(config)
+	if err != nil {
+		Logf("Unable to load volume provisioner client %q: %v", config, err)
+	}
+	return blockStorageClient
+}
+
+func (f *Framework) newConfigurationProvider(cfg *client.Config) (common.ConfigurationProvider, error) {
+	var conf common.ConfigurationProvider
+	if cfg != nil {
+		err := cfg.Validate()
+		if err != nil {
+			return nil, errors.Wrap(err, "invalid client config")
+		}
+		if cfg.UseInstancePrincipals {
+			Logf("Using instance principals configuration provider")
+			cp, err := auth.InstancePrincipalConfigurationProvider()
+			if err != nil {
+				return nil, errors.Wrap(err, "InstancePrincipalConfigurationProvider")
+			}
+			return cp, nil
+		}
+		Logf("Using raw configuration provider")
+		conf = common.NewRawConfigurationProvider(
+			cfg.Auth.TenancyOCID,
+			cfg.Auth.UserOCID,
+			cfg.Auth.Region,
+			cfg.Auth.Fingerprint,
+			cfg.Auth.PrivateKey,
+			common.String(cfg.Auth.PrivateKeyPassphrase))
+	} else {
+		conf = common.DefaultConfigProvider()
+	}
+	return conf, nil
+}
+
+// CheckEnvVar checks if an environment variable is set in the werker environement, if not it checks the test context.
+func (f *Framework) CheckEnvVar(envVar string) string {
```

Comment:
For now if we could replace all env variables with flags and properties on `TestContext`. As a follow on it would be great to get it set up so that you could use either flags or env vars but I think we shouldn't mix and match in a patchy fashion.